lib/document.js

```diff
@@ -1333,10 +1333,11 @@ Document.prototype.$set = function $set(path, val, type, options) {
   // When using the $set operator the path to the field must already exist.
   // Else mongodb throws: "LEFT_SUBFIELD only supports Object"
 
-  if (parts.length <= 1) {
+  if (parts.length < 2) {
```

Comment:
Is < faster than <=? If so, I'll make sure to use that going forward

Reply:
This is an artifact. 
```suggestion
  if (parts.length <= 1) {
```